### PR TITLE
Fix schema for CRDs without validation

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go [hunk 1]
```diff
@@ -390,18 +388,15 @@ func newBuilder(crd *apiextensions.CustomResourceDefinition, version string, sch
...
-		})
+
+	if schema == nil {
+		schema = &spec.Schema{
+			SchemaProps: spec.SchemaProps{Type: []string{"object"}},
+		}
 	}
+
+	// Pre-build schema with Kubernetes native properties
+	b.schema = b.buildKubeNative(schema)
```
Comment (roycaihw) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go:399]: adding fields to empty schema closes the door for unknown fields during client-side validation

we cannot do this before we honor `additionalProperties` in kube-openapi and kubectl

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go:399]: if we add `additionalProperties: true` explicitly, that would work, right?

But very good observation. Thanks for noticing that.

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go:399]: Which reminds me again that we have to stop using heuristics to make specs stricter. We can leave out interpretations, but never ever be stricter than OpenAPI is officially.

Comment (roycaihw) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go:399]: my understanding is that we don't honor `additionalProperties: true` in this [code path](https://github.com/kubernetes/kube-openapi/blob/ea82251f3668f8c1bde607fa6e20e5bf36e576a4/pkg/util/proto/document.go#L237-L238)

> if we add `additionalProperties: true` explicitly, that would work, right?

I will double check

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go:399]: We don't. I checked already.

**Comment by sttts** [comment]
~~Was this change lost in the PR branch? https://github.com/kubernetes/kubernetes/pull/71192/files#diff-3c4a34afddbbbdd10ef27fb056228fd9R35~~

nvmd. Wrong PR.

**Pull request merged**